client/client_test.go

```diff
@@ -551,6 +560,89 @@ func withToken(t *testing.T, asUser *client.Client) *client.Client {
 	return asUser
 }
 
+var tearDownResource = func(pool *dockertest.Pool, r *dockertest.Resource) error {
+	var err error
+	if pool == nil {
+		pool, err = setupDockerPool()
+		if err != nil {
+			return err
+		}
+	}
+	err = pool.Purge(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setupCoreInstance(pool *dockertest.Pool, baseURL string, token types.Token) (*dockertest.Resource, error) {
```

Comment:
Let's see. This new function is not being called on TestMain.
I think you want to initilize it only on the necessary tests 🤔 
What if we make docker pool a package variable and we reuse it.
That way you don't need to cleanup it up after using it.